python/lsst/ts/envsensors/command_handler.py

```diff
@@ -215,6 +216,23 @@ def _validate_device_configuration(self, device_configuration: Dict[str, Any]):
                     response_code=ResponseCode.INVALID_CONFIGURATION,
                 )
 
+        # Make sure that Key.CHANNELS is present for SensorType.TEMPERATURE
+        # devices and not for others.
+        if device_configuration[Key.SENSOR_TYPE] == SensorType.TEMPERATURE:
+            if Key.CHANNELS not in device_configuration:
+                raise CommandError(
+                    msg=f"Missing configuration key {Key.CHANNELS} for sensor of "
+                    f"type {SensorType.TEMPERATURE}",
+                    response_code=ResponseCode.INVALID_CONFIGURATION,
+                )
+        else:
+            if Key.CHANNELS in device_configuration:
+                raise CommandError(
+                    msg=f"Configuration key {Key.CHANNELS} should not be present "
+                    f"for sensor of type {device_configuration[Key.SENSOR_TYPE]}",
+                    response_code=ResponseCode.INVALID_CONFIGURATION,
+                )
```

Comment:
This does not scale well as you add sensor types. Please consider a more general solution.

One approach:
* Make sure self.required_keys is minimal. These keys must be present for all sensor types. (I"m guessing you are there).
* Make an "extra keys" dict of SensorType: (required_keys, optional_keys).

Then for each sensor type you can quickly compute the full set of required keys and allowed keys and check the configuration against that.

Reply:
I am struggling with a way to implement this efficiently. There are several optional keys and I see now that there are some checks missing. I think a better way would be to use a json schema and check against that. I created a ticket for this and added a `TODO` referring to the ticket.